src/views/connectivity/ConnectionDetail/index.tsx

```diff
@@ -481,24 +481,24 @@ const ConnectionDetail = withStyles(styles)(
                     mandatory={mandatory}
                     isCreateParameter={isAddingParameter}
                     onEdit={(newParameter) => {
-                        const newParameters = [...newConnectionDetail.environments[environmentIndex].parameters];
+                        const newParameters = isAddingParameter
+                            ? [...newConnectionDetail.environments[environmentIndex].parameters, newParameter]
+                            : [...newConnectionDetail.environments[environmentIndex].parameters];
```

Comment:
you're mixing editing a parameter and adding a new parameter. Either you modify an existing one OR you add a new one, not both